[Repository Rewrite] Test for disabling rh repository with basearch

ColeHiggins2 opened the pull request:
Test for disabling a red hat repository with basearch set

BZ: 1932486

Steps:
1.  You have the Appstream Kicstart repositories release version "8" synced in from the release of RHEL 8
2.  hammer repository-set disable --basearch --name --product-id --organization --releasever

sambible commented on tests/foreman/cli/test_repositories.py:
```diff
@@ -47,3 +49,50 @@ def test_positive_custom_products_disabled_by_default(
+    )
+    repo = module_target_sat.api.Repository(id=rh_repo_id).read()
+    repo.sync(timeout=2000)
+    module_target_sat.execute(
```
Is this line just to basically assert that the repo was synced correctly?

sambible replied:
to be clear, I'm referring to the repo list call after line 85

Griffin-Sullivan replied:
If so we should add an assertion

sambible reviewed (approved):
Looks good, just curious about that one piece

Griffin-Sullivan commented:
trigger: test-robottelo
pytest: tests/foreman/cli/test_repositories.py::test_positive_test_disable_rh_repo_with_basearch

Griffin-Sullivan reviewed (approved):
Ack pending comment

shubhamsg199 commented on tests/foreman/cli/test_repositories.py:
```diff
@@ -47,3 +49,50 @@ def test_positive_custom_products_disabled_by_default(
     assert 'Enabled:   0' in product_details.stdout
+
+
+def test_positive_test_disable_rh_repo_with_basearch(
```
```suggestion
def test_positive_disable_rh_repo_with_basearch(
```

ColeHiggins2 commented:
@Gauravtalreja1 @Griffin-Sullivan @sambible @shubhamsg199  Addressed comments
1. Updated test name
2. Removed extra sat.execute command(no longer needed)
3. Updated to RepositorySets.disabled

Gauravtalreja1 commented on tests/foreman/cli/test_repositories.py:
```diff
@@ -47,3 +49,45 @@ def test_positive_custom_products_disabled_by_default(
+    disabled_repo = module_target_sat.cli.RepositorySet.disable(
+        {
+            'basearch': DEFAULT_ARCHITECTURE,
+            'name': "Red Hat Enterprise Linux 8 for x86_64 - BaseOS (Kickstart)",
```
```suggestion
            'name': REPOSET['kickstart']['rhel8'],
```

Gauravtalreja1 commented on tests/foreman/cli/test_repositories.py:
```diff
@@ -47,3 +49,45 @@ def test_positive_custom_products_disabled_by_default(
+        releasever=REPOS['kickstart']['rhel8_aps']['version'],
+    )
+    repo = module_target_sat.api.Repository(id=rh_repo_id).read()
+    repo.sync(timeout=2000)
```
isn't this timeout too much time for kickstart repo? 
```suggestion
    repo.sync(timeout=600)
```

Gauravtalreja1 commented on tests/foreman/cli/test_repositories.py:
```diff
@@ -47,3 +49,45 @@ def test_positive_custom_products_disabled_by_default(
+            'name': "Red Hat Enterprise Linux 8 for x86_64 - BaseOS (Kickstart)",
+            'product-id': repo.product.id,
+            'organization-id': module_entitlement_manifest_org.id,
+            'releasever': 8,
```
```suggestion
            'releasever': REPOS['kickstart']['rhel8_aps']['version'],
```

Gauravtalreja1 reviewed (commented):
Ack, with a few suggestions

ColeHiggins2 commented:
trigger: test-robottelo
pytest: tests/foreman/cli/test_repositories.py::test_positive_test_disable_rh_repo_with_basearch

ColeHiggins2 commented:
trigger: test-robottelo
pytest: tests/foreman/cli/test_repositories.py::test_positive_disable_rh_repo_with_basearch

damoore044 reviewed (approved):
looks good! prt pending
